ManageBookings PR

wkonit opened the pull request:
### What this PR does (required):
Made the UI for the manage bookings page. Still needs some tweaking.

### Screenshots / Videos (front-end only):
![Screenshot 2022-02-22 182002](https://user-images.githubusercontent.com/98362157/155237732-31def6a3-841f-41b6-a3c7-aa573f37ccba.png)

### Any information needed to test this feature (required):
- Post what steps to take to be able to test this feature (eg. "sign up as a new user and upload a profile picture from the navbar")
- Need to be logged in then navigate the the managebookings page via the url (eg: http://localhost:3000/managebookings)
- Need to install dependencies such as: 
  npm install @mui/material @emotion/react @emotion/styled
  npm install @mui/lab
  npm install date-fns --save

### Any issues with the current functionality (optional):
- List any problems where you were not able to complete all tasks on ticket
  I was not able to figure out to set the CurrentBookings and the PastBookings part yet

- List any problems this PR may cause for the project or other active PRs
  Since I messed up with how I cloned the repo, there might have to be a couple of merges to do?

- Post a screenshot/video of the problem, along with detailed console outputs where applicable

moffatethan commented on client/src/pages/ManageBookings/CalenderView.tsx:
```diff
@@ -0,0 +1,92 @@
+import ArrowForwardIosIcon from '@mui/icons-material/ArrowForwardIos';
+
+
+// const minDate = new Date('2020-01-01T00:00:00.000');
+// const maxDate = new Date('2034-01-01T00:00:00.000');
```
Let's remove this 😄

moffatethan commented on client/src/pages/ManageBookings/CalenderView.tsx:
```diff
@@ -0,0 +1,92 @@
+
+    function dayBGColor(day:Date){
+      if(!isSameMonth(day, selectedDate)) return '#FFFFFF';
+      if(isSameDay(day, selectedDate)) return '#f14140';
```
Okay, so you have the right idea but the red background is actually showing days they have bookings available

moffatethan commented on client/src/pages/ManageBookings/CurrentBookings.tsx:
```diff
@@ -0,0 +1,66 @@
+
+import useStyles from './useStyles';
+
+import Avatar from '@mui/material/Avatar';
+
+
+
+
+const CurrentBookings = () => {
```
Lots of whitespace here, could we clean that up?

moffatethan commented on client/src/pages/ManageBookings/CurrentBookings.tsx:
```diff
@@ -0,0 +1,66 @@
+    return (
+        <div>
+            <Card sx={{width: 500, height: 200}}>
+                
+                <CardHeader
+                
+                    action={
+                        <IconButton>
+                            <SettingsIcon/>
+                        </IconButton>
+                    }
```
Same here

moffatethan commented on client/src/pages/ManageBookings/CurrentBookings.tsx:
```diff
@@ -0,0 +1,66 @@
+                        </IconButton>
+                    }
+                    titleTypographyProps={{ variant: "subtitle2", pb: 2, sx: {fontWeight: 600, fontFamily: "Arial"}}}
+                    title = "YOUR NEXT BOOKING"
+                    subheaderTypographyProps={{ variant:"h6", sx: {fontWeight: 600, color:"black"}}}
+                    subheader = {dummyBookings.map((dummyBooking) => dummyBooking.date)}
```
Could we not have space between the equals

moffatethan commented on client/src/pages/ManageBookings/ManageBookings.tsx:
```diff
@@ -0,0 +1,30 @@
+            <Grid container> 
+                <Grid columns={1} justifyContent="flex-start" className={classes.GridTop}>
+                    <NextBooking/>
+                    {/* <CurrentBookings/> */}
```
Let's remove this

moffatethan commented on client/src/pages/ManageBookings/NextBooking.tsx:
```diff
@@ -0,0 +1,54 @@
+
+
+
+const NextBooking = () => {
+    const classes = useStyles()
+    return (
+        <div>
```
So, thinking in React. This probably isn't the best approach, the current bookings and next booking style will be **identical** so we should just make one component but then change the props for each booking but the styling stays the same so we can have this be more maintainable and keep the codebase concise.

moffatethan commented on client/src/pages/ManageBookings/useStyles.ts:
```diff
@@ -0,0 +1,19 @@
+
+const useStyles = makeStyles((theme: Theme) => {
+    return{
+        GridTop:{
+            paddingTop: theme.spacing(6)
+        },
+        Avatar:{
+            float: "left"
+        },
+        CalenderWeek:{
```
Let's do camelCase for the CSS names

wkonit commented:
![ManageBookings 3-5-2022](https://user-images.githubusercontent.com/98362157/156878185-7a5dc902-e8ec-4374-b11e-6e85e8d57e26.png)

New Changes:
- There is test data in `ManageBookings.tsx` to help with mapping the bookings that can be removed once backend integration is done. 
- Currently the gear button switches between `ACCEPTED` AND `DECLINED` when pressed. Can be improved later and icon can be switched out with something better.
- The selected date on the `CalenderView` is currently set to the current day. There is a variable ready to be set for different date. When booking schedules are more clear, the variable can be edited to take more days but it is only set to take one date now. 
- Removed the seperate booking components leaving only one. 
- Removed `useStyles.ts`.

Issues:
- Couldn't figure out why the single digits dates not being centered despite `alignItems:"center"` and` justifyContent:"center"`.
- Currently there is an error where I cannot use the url to access` /managebookings`. Instead I will be constantly redirected to the login page. I bypassed this issue by routing the login page to `/managebookings`. I'm not sure what's causing the issue, hopefully someone can look into it?

Questions:
- I'm not sure how to really edit the scrollbar so it's just the standard windows(?) one. 
- I'm not 100% sure how to resolve the conflicting files for `client/src/App.tsx`.
